pkg/network/kubemacpool.go

```diff
@@ -69,6 +68,22 @@ func renderKubeMacPool(conf *opv1alpha1.NetworkAddonsConfigSpec, manifestDir str
 		conf.KubeMacPool.RangeEnd = rangeEnd.String()
 	}
 
+	return []error{}
+}
+
+func changeSafeKubeMacPool(prev, next *opv1alpha1.NetworkAddonsConfigSpec) []error {
+	if prev.KubeMacPool != nil && !reflect.DeepEqual(prev.KubeMacPool, next.KubeMacPool) {
+		return []error{errors.Errorf("cannot modify KubeMacPool configuration once it is deployed")}
+	}
+	return nil
```

Comment:
Oh I see it was used before. Wonder how that works with append

Reply:
Oh yes, we should, otherwise we endup with []error{<nil>}. If you don't mind

Reply:
Done :) can you please take a look?